src/metabase/util.cljc

```diff
@@ -917,3 +917,31 @@
               (map-all f (rest s1) (rest s2)))))))
   ([f c1 c2 & colls]
    (map-all* f (list* c1 c2 colls))))
+
+(defn row-diff
+  "Given two vectors of row maps - `existing` is what is in the database, `incoming` is what you want to see there,
+  returns a map of 4 lists: `{to-skip, to-delete, to-update, to-insert}`. Executing on those lists will bring database
```

Comment:
the first time I read this I thought it might return the IDs, so it might be worth clarifying these are just groups of rows
```suggestion
  returns a map of 4 lists of rows: `{to-skip, to-delete, to-update, to-insert}`. Executing on those lists will bring the database
```